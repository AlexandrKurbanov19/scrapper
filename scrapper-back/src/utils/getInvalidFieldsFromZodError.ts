import { ZodError } from 'zod';

const getInvalidFieldsFromZodError = (zodError: ZodError) => (
  zodError.issues
    .map((issue) => issue.path.join('.'))
);

export default getInvalidFieldsFromZodError;
