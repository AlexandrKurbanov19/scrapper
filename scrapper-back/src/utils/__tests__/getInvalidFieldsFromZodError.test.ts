import argsSchema from '../../custom-graphql/mutation/registration/argsSchema';
import getInvalidFieldsFromZodError from '../getInvalidFieldsFromZodError';

describe('getInvalidFieldsFromZodError', () => {
  it('can get error list', () => {
    const validationResult = argsSchema.safeParse({input: {}});

    expect(validationResult.success).toBe(false);

    if (validationResult.success === false) {
      expect(getInvalidFieldsFromZodError(validationResult.error)).toMatchSnapshot();
    }
  });
});
