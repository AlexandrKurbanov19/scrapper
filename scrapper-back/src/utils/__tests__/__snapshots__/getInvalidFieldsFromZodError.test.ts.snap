// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getInvalidFieldsFromZodError can get error list 1`] = `
[
  "input.phone",
  "input.fio",
  "input.birthday",
]
`;
